sympy/combinatorics/permutations.py

```diff
@@ -1243,8 +1243,11 @@ def __mul__(self, other):
         Cycle(1, 3, 2)
 
         """
+        from sympy.combinatorics.perm_groups import PermutationGroup
         a = self.array_form
         # __rmul__ makes sure the other is a Permutation
+        if isinstance(other, PermutationGroup):
+            return [a*i for i in other.generators]
```

Comment:
Cosets are not groups in general (they usually don't even have the identity). It would be nice to have some special object here, since the set of all cosets from a normal subgroup themselves form a group (a so-called quotient group). But I don't know if we need that for this pull request (unless this object already exists).